strawberry/schema/types/base_scalars.py

```diff
@@ -21,6 +21,13 @@ def inner(value: str):
     return inner
 
 
+def parse_decimal(value: str):
+    try:
+        return decimal.Decimal(value)
+    except decimal.DecimalException:
+        raise GraphQLError(f'Value cannot represent a Decimal: "{value}".')
```

Comment:
We should consider making this string a common asset that both `wrap_parser` and `parse_decimal` use

Reply:
I didn't want to reuse the same format string because the formatted decimal exceptions aren't very API friend (ex: `[<class 'decimal.ConversionSyntax'>]`).

I could modify `wrap_parser` to something like the following if you wanted:

```python
from typing import Callable

from graphql import GraphQLError


def wrap_parser(
    parser: Callable,
    type_: str,
    classinfo: type = ValueError,
    format_exception: bool = True,
) -> Callable:
    def inner(value: str):
        try:
            return parser(value)
        except Exception as e:
            if not isinstance(e, classinfo):
                raise
            message = f'Value cannot represent a {type_}: "{value}".'
            if format_exception:
                message += f" {e}"
            raise GraphQLError(message)

    return inner


date_parser = wrap_parser(datetime.date.fromisoformat, "Date")
decimal_parser = wrap_parser(
    decimal.Decimal, "Decimal", decimal.DecimalException, False
)
```

Reply:
Nah if it's inconvenient, let's not worry about it :+1: